nixos/modules/services/web-apps/ente.md

```diff
@@ -0,0 +1,181 @@
+    # your desired minio root user and password.
+    #
+    # MINIO_ROOT_USER=your_root_user
+    # MINIO_ROOT_PASSWORD=a_randomly_generated_long_password
+    rootCredentialsFile = "/run/secrets/minio-credentials-full";
+  };
+
+  systemd.services.minio.environment.MINIO_SERVER_URL = "https://s3.example.com";
+
+  # Proxy for minio
+  networking.firewall.allowedTCPPorts = [
+    80
+    443
+  ];
+  services.nginx = {
+    recommendedProxySettings = true;
+    virtualHosts."s3.example.com" = {
+      forceSSL = true;
+      useACME = true;
+      locations."/".proxyPass = "http://localhost:9000";
+      # determine max file upload size
+      extraConfig = ''
+        client_max_body_size 16G;
+        proxy_buffering off;
+        proxy_request_buffering off;
+      '';
+    };
+  };
+}
+```
+
+And the configuration for ente:
+
+```nix
+{
+  services.ente = {
+    web = {
+      enable = true;
+      domains = {
+        accounts = "accounts.example.com";
+        albums = "albums.example.com";
+        cast = "cast.example.com";
+        photos = "photos.example.com";
+      };
+    };
+    api = {
+      enable = true;
+      nginx.enable = true;
+      # Create a local postgres database and set the necessary config in ente
+      enableLocalDB = true;
+      domain = "api.example.com";
+      # You can hide secrets by setting xyz._secret = file instead of xyz = value.
+      # Make sure to not include any of the secrets used here directly
+      # in your config. They would be publicly readable in the nix store.
+      # Use agenix, sops-nix or an equivalent secret management solution.
+      settings = {
+        s3 = {
+          use_path_style_urls = true;
+          b2-eu-cen = {
+            endpoint = "https://s3.example.com";
+            region = "us-east-1";
+            bucket = "b2-eu-cen";
+            key._secret = pkgs.writeText "minio_user" "minio_user";
+            secret._secret = pkgs.writeText "minio_pw" "minio_pw";
+          };
+        };
+        key = {
+          # generate with: openssl rand -base64 32
+          encryption._secret = pkgs.writeText "encryption" "T0sn+zUVFOApdX4jJL4op6BtqqAfyQLH95fu8ASWfno=";
+          # generate with: openssl rand -base64 64
+          hash._secret = pkgs.writeText "hash" "g/dBZBs1zi9SXQ0EKr4RCt1TGr7ZCKkgrpjyjrQEKovWPu5/ce8dYM6YvMIPL23MMZToVuuG+Z6SGxxTbxg5NQ==";
+        };
+        # generate with: openssl rand -base64 32
+        jwt.secret._secret = pkgs.writeText "jwt" "i2DecQmfGreG6q1vBj5tCokhlN41gcfS2cjOs9Po-u8=";
+      };
+    };
+  };
+
+  networking.firewall.allowedTCPPorts = [
+    80
+    443
+  ];
+  services.nginx = {
+    recommendedProxySettings = true; # This is important!
+    virtualHosts."accounts.${domain}".enableACME = true;
+    virtualHosts."albums.${domain}".enableACME = true;
+    virtualHosts."api.${domain}".enableACME = true;
+    virtualHosts."cast.${domain}".enableACME = true;
+    virtualHosts."photos.${domain}".enableACME = true;
+  };
```

Comment:
Not sure if it would be worth to add an option for this 🤔

Reply:
If you need a data point, I have a custom NixOS module in my config so I always do `configureNginx = false;` for services, and my custom module allows me to just do `services.nginx.subdomains.mysubdomain.port = 9876;` (with some other options available too but you get the idea).

All my services use `useACMEHost`, not `enableACME`.

My personal opinion is to just leave NGINX up to the user to configure, because usually I want something slightly different from what's available via the service's own options.